h2o-core/src/main/java/water/server/ServletMeta.java

```diff
@@ -9,10 +9,21 @@
 
   private final String _contextPath;
   private final Class<? extends HttpServlet> _servletClass;
+  private final boolean _alwaysEnabled;
 
-  public ServletMeta(String contextPath, Class<? extends HttpServlet> servletClass) {
+  /**
+   * Constructs a new instance of {@link ServletMeta} with the alwaysEnabled functionality either turned on or off.
+   * An always enabled servlet is guaranteed to be active on every node, even if the rest of the API is shut down
+   * on a particular node. This is useful for vital, per-node APIs. Typically used by non user-facing functionality.
+   * 
+   * @param contextPath Context path the underlying servlet handles
+   * @param servletClass Specific implementation of the {@link HttpServlet} to handle the context path
+   * @param alwaysEnabled Set to true if this servlet should remain active even the API is turned off on a particular node.
+   */
+  public ServletMeta(final String contextPath, final Class<? extends HttpServlet> servletClass, final boolean alwaysEnabled) {
```

Comment:
This is adjusted mainly due to Sparkling Water using some endpoints for data upload on each node which must absolutely always be enabled, and especially on Kubernetes :wink: 

I was talking to @mn-mikke about this and we both agreed SW should be in control which endpoints remain active and it should be immediately clear from the code. The alternative to this is list the allowed context paths as Strings on H2O's side, but that would easily go out of sync with SW.